### PR TITLE
Modularize eslintrc, add tests

### PR DESCRIPTION
**Pull request by justjake**
- modularize eslint-config-airbnb a little bit: 
  - `eslint-config-airbnb/base` gives a react-free version
  - `eslint-config-airbnb` still gives the react-loaded version
- add tests to eslint-config-airbnb
  - validate that the React order rule config works
  - validate that `eslint-config-airbnb/base` does not depend on React plugin

I would consider making `eslint-config-airbnb/base` the default export, and making `eslint-config-airbnb/react` contain the react rules.

Let me know what you think.

/cc @ljharb @hshoff

### CODE REVIEW

**Comment by justjake** [comment]
I use `babel-tape-runner` because multiline strings are so much nicer in ES6.

Unfortunately I'm not linting the tests because `eslint` can't use `.js` config files unless its loading them from an NPM module :(

**Comment by ljharb** [comment]
Assuming none of the rules were changed, merely moved, this is awesome! Love the use of `tape`, and `babel-tape-runner` is just fine imo. :shipit:

(In other words, the config is already in JS instead of JSON, this is just improving the organization. the default primary export is identical, so, only a patch change. well done)

**Comment by hshoff** [comment]
han-approved

File: packages/eslint-config-airbnb/react.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    'react/no-unknown-property': 2,  // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unknown-property.md
+    'react/prop-types': 2,           // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md
+    'react/react-in-jsx-scope': 2,   // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md
+    'react/self-closing-comp': 2,    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/self-closing-comp.md
+    'react/wrap-multilines': 2,      // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/wrap-multilines.md
+    'react/sort-comp': [2, {         // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md
+      'order': [
+        'lifecycle',
+        '/^on.+$/',
+        '/^(get|set)(?!(InitialState$|DefaultProps$|ChildContext$)).+$/',
```
Comment (justjake) [inline packages/eslint-config-airbnb/react.js]: @ljharb @kasperekt this PR also fixes https://github.com/airbnb/javascript/pull/418 -- and I've added tests to make sure this regex works, and that it does not regress.

**Comment by ljharb** [comment]
:+1: LGTM!

**Comment by kruppel** [comment]
@justjake any plans to bump/republish the eslint-config-airbnb in npm?

**Comment by justjake** [comment]
ok

**Comment by justjake** [comment]
@kruppel done

**Comment by kruppel** [comment]
> @kruppel done

Thanks, good stuff btw.

**Pull request merged**